operators/src/processing/raster_vector_join/non_aggregated.rs

```diff
@@ -166,79 +178,84 @@ impl PointRasterJoiner {
             })
             .unzip();
 
-        let mut valid = vec![false; points.len()];
+        let mut valid = vec![false; collection.len()];
         for i in indexes {
             valid[i] = true;
         }
 
-        let points = points.filter(valid)?;
-        let points = points.replace_time(&time_intervals)?;
+        let collection = collection.filter(valid)?;
+        let collection = collection.replace_time(&time_intervals)?;
 
-        self.state = Some(JoinerState {
-            aggregator: create_feature_aggregator::<P>(points.len(), self.aggregation_method),
-            points,
+        self.state = Some(JoinerState::<G, C> {
+            aggregator: create_feature_aggregator::<P>(collection.len(), self.aggregation_method),
+            covered_pixels: collection.create_covered_pixels(),
+            g: Default::default(),
         });
 
         Ok(())
     }
 
     fn extract_raster_values<P: Pixel>(
         mut self,
-        points: &MultiPointCollection,
+        initial_collection: &FeatureCollection<G>,
         raster: &RasterTile2D<P>,
     ) -> Result<Self> {
         let state = loop {
             if let Some(state) = &mut self.state {
                 break state;
             }
 
-            self.initialize::<P>(points, &raster.time)?;
+            self.initialize::<P>(initial_collection, &raster.time)?;
         };
-        let points = &state.points;
+        let collection = &state.covered_pixels.collection_ref();
         let aggregator = &mut state.aggregator;
+        let covered_pixels = &state.covered_pixels;
 
-        // TODO: avoid looking up coordinates not contained in the raster
-
-        for (index, feature) in points.into_iter().enumerate() {
-            for &coordinate in feature.geometry.points() {
-                let value = match raster.pixel_value_at_coord(coordinate) {
+        for feature_index in 0..collection.len() {
+            for grid_idx in covered_pixels.covered_pixels(feature_index, raster) {
+                let value = match raster.get_at_grid_index(grid_idx) {
                     Ok(value) => value,
                     Err(_) => continue, // not found in this raster tile
```

Comment:
we need to check if it is an out of bounds error, else return error (or ignore)? see other comment

Reply:
cf my other comment